### PR TITLE
67 improve types

### LINKED ISSUE #67: Improve types and export relevant ones
For example the `Token` type can be improved, instead of:

```typescript
interface Token {
  access_token: string;
  expires_in: number;
  scope: string;
  token_type: string;
}
```

we should use something like: 

```typescript
interface Token {
  access_token: string
  expires_in?: number
  scope?: string
  token_type?: string
  [key: string]: {}
}
```

since we do not know what properties are exactly set on the tokens in different implementations. Furthermore, the `Token` type should be exported by the library. For the types`Token` and `Tokeninfo` we should check again whether it is necessary to have two types or if one (which is than more generic) is enough.

### PR DESCRIPTION
**Pull request by bzums**
* get rid of the redundant type `TokenInfo` and use `Token` instead
* export relevant types (like `Token`)
* a number of minor refactorings and doc adaptions

close #67

### CODE REVIEW

File: src/TokenCache.ts [hunk 1]
```diff
@@ -87,21 +79,14 @@ class TokenCache {
...
-          .then((tokenInfo: TokenInfo) => {
+          .then((token) => getTokenInfo(this.oauthConfig.tokenInfoEndpoint, token.access_token))
+          .then((token) => {
 
-            tokenInfo.local_expiry = Date.now() + tokenInfo.expires_in * 1000 - EXPIRE_THRESHOLD;
-            this._tokens[tokenName] = tokenInfo;
-            resolve(tokenInfo);
+            token.local_expiry = Date.now() + token.expires_in * 1000 - EXPIRE_THRESHOLD;
+            this._tokens[tokenName] = token;
+            return Promise.resolve(token);
```
Comment (ISO50) [inline src/TokenCache.ts]: Or just return token. Would leave a blank line to be aligned with our coding guidelines.

Comment (bzums) [inline src/TokenCache.ts]: 👍 will tackle this

**Comment by ISO50** [comment]
👍

**Comment by Retro64** [comment]
:-1:

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lib-oauth-tooling",
-  "version": "2.2.1",
+  "version": "2.3.0",
```
Comment (Retro64) [inline package.json]: As TokenInfo is obsolete and might be used by consuming programs, version should switch to 3.0.0

Comment (bzums) [inline package.json]: `TokenInfo` was not exported, so can not be used by consuming code.

Comment (bzums) [inline package.json]: Nevertheless I would like to test the version in the wild before publishing it to make sure nothing is broken.

File: integration-test/mock-tooling/index.ts [hunk 3]
```diff
@@ -83,17 +82,17 @@ describe('mock tooling', () => {
 
       // when
       const promise = getTokenInfo(tokeninfoEndpoint, 'foo')
-      .then((token: TokenInfo) => {
+      .then((token) => {
```
Comment (Retro64) [inline integration-test/mock-tooling/index.ts]: Types?

**Comment by Retro64** [comment]
:+1:

**Comment by bzums** [comment]
👍

**Pull request merged**